### PR TITLE
perf(utils): further optimise subset check.

### PR DESCRIPTION
**Pull request by yuroitaki**
As https://github.com/tlsnotary/tlsn/pull/664 starts using `is_subset` checks, I implemented a small optimisation — immediately return false if the subset candidate's left and right boundaries are not within/same as the other rangeset.

### CODE REVIEW

**Review by themighty1 — APPROVED** [review]
lgtm

**Comment by sinui0** [comment]
Lgtm, but performance optimizations would ideally come with a benchmark proving they are faster

**Comment by yuroitaki** [comment]
> Lgtm, but performance optimizations would ideally come with a benchmark proving they are faster

do you mean adding a `bench.rs` in the code, or attaching the result of manual benching in this PR?

**Comment by yuroitaki** [comment]
Benchmark results after optimisation in my m2 max macbook below:

image

TLDR: 
(1) No impact in the worst case for subset check of range against rangeset.
(2) Significant improvement in the worst case for subset check between rangesets.

(1) most likely because each of the iteration saved by the short-circuit optimisation only does very little work, i.e. `continue`, which doesn't amount to any noticeable performance improvement.

**Review by sinui0 — APPROVED** [review]
